### ISSUE TITLE
Input component in vue's RadListView template breaks app

### ISSUE THREAD

**Issue by wendt88** [telerik/nativescript-ui-feedback#966]
relevant packages:
```
tns-android: "5.1.0"
nativescript-ui-listview: "^5.0.1"
nativescript-vue: "^2.0.0"
```

Only on android, if there is some Input-component (like `TextField`, `Switch`, `TextView`) inside the RadListView template, during scroll, the app breaks with the current error: 
```
System.err: com.tns.NativeScriptException:
System.err: Calling js method onCreateViewHolder failed
System.err:
System.err: TypeError: Cannot read property 'push' of null
System.err: File: "file:///[path]/files/app/vendor.js, line: 64287, column: 24
System.err:
System.err: StackTrace:
System.err:     Frame: function:'ListViewAdapter.onCreateViewHolder', file:'file:///[path]/files/app/vendor.js', line: 64287, column: 25
System.err:
System.err:     at com.tns.Runtime.callJSMethodNative(Native Method)
System.err:     at com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1120)
System.err:     at com.tns.Runtime.callJSMethodImpl(Runtime.java:1000)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:987)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:967)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:959)
System.err:     at com.tns.gen.com.telerik.widget.list.ListViewAdapter_vendor_64216_26_ListViewAdapter.onCreateViewHolder(ListViewAdapter_vendor_64216_26_ListViewAdapter.java:23)
System.err:     at com.telerik.widget.list.ListViewWrapperAdapter.onCreateViewHolder(ListViewWrapperAdapter.java:468)
System.err:     at com.telerik.widget.list.ListViewWrapperAdapter.onCreateViewHolder(ListViewWrapperAdapter.java:20)
System.err:     at android.support.v7.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:6794)
System.err:     at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5975)
System.err:     at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5858)
System.err:     at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5854)
System.err:     at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2230)
System.err:     at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1557)
System.err:     at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1517)
System.err:     at android.support.v7.widget.LinearLayoutManager.scrollBy(LinearLayoutManager.java:1331)
System.err:     at android.support.v7.widget.LinearLayoutManager.scrollVerticallyBy(LinearLayoutManager.java:1075)
System.err:     at android.support.v7.widget.RecyclerView.scrollStep(RecyclerView.java:1832)
System.err:     at android.support.v7.widget.RecyclerView$ViewFlinger.run(RecyclerView.java:5067)
System.err:     at android.view.Choreographer$CallbackRecord.run(Choreographer.java:966)
System.err:     at android.view.Choreographer.doCallbacks(Choreographer.java:778)
System.err:     at android.view.Choreographer.doFrame(Choreographer.java:710)
System.err:     at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:952)
System.err:     at android.os.Handler.handleCallback(Handler.java:789)
System.err:     at android.os.Handler.dispatchMessage(Handler.java:98)
System.err:     at android.os.Looper.loop(Looper.java:164)
System.err:     at android.app.ActivityThread.main(ActivityThread.java:6798)
System.err:     at java.lang.reflect.Method.invoke(Native Method)
System.err:     at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
```

Here is a playground wich causes the same: https://play.nativescript.org/login?template=play-vue&id=n0x0DG

**Comment by zbranzov** [comment]
My observation led me to fact that the issue is caused by the v-model. If you add it to the label and remove the rest of the elements it will also break the app on android and shuffle the bound value on iOS.

**Comment by tbozhikov** [comment]
@zbranzov The crash you get after the fix is caused by Vue updating the underlying data object. 

1. RLV loads all items initially, creates containers only for the visible ones
2. (when scrolled) RLV begins to reuse the containers that disappear from the screen, changing only the data context to the new data item
- because v-model (vue's two-way) binding is used, when the RLV assigns a new data item to the container, Vue thinks the value of the previously underlying data item is changed
- the above triggers refresh() call in the RLV, which, eventually, triggers crash

This can be avoided if we give a hint to Vue about each item's identity. See the [docs on `key`](https://vuejs.org/v2/guide/list.html#key). Below is a working code snippet:

```vue
<template>
    <Page class="page">
        <ActionBar title="Home" class="action-bar" />
        <StackLayout>
            <RadListView for="item in items">
                <v-template>
                    <StackLayout class="p-10" :key="item.id">
                        <Label :text="item.id" />
                        <Label :text="item.name" />
                        <Label v-model="item.bool" /> 
                        <Switch v-model="item.bool" />
                    </StackLayout>
                </v-template>
            </RadListView>
        </StackLayout>
    </Page>
</template>

<script>
    export default { 
        data() {
            let items = [];
            for (let i = 1; i <= 20; i++) {
                items.push({
                    id: i,
                    name: `Item ${i}`,
                    bool: !(i % 2)
                });
            }
            return {
                items
            };
        }
    };
</script>
```

**Comment by zbranzov** [comment]
The fix is released in nativescript-ui-listview@6.0. Notice that the versions contains breaking change https://github.com/telerik/nativescript-ui-feedback/blob/master/releases/listview.md
